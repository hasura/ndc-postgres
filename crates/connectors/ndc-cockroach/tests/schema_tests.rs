pub mod common;

#[tokio::test]
async fn get_schema() {
    let result = tests_common::request::get_schema(common::create_router().await).await;
    insta::assert_json_snapshot!(result);
}
