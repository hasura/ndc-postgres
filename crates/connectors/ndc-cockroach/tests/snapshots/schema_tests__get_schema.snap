---
source: crates/connectors/ndc-cockroach/tests/schema_tests.rs
expression: result
---
{
  "scalar_types": {
    "bigint": {
      "aggregate_functions": {},
      "comparison_operators": {
        "_eq": {
          "argument_type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "_gt": {
          "argument_type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "_gte": {
          "argument_type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "_lt": {
          "argument_type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "_lte": {
          "argument_type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "_neq": {
          "argument_type": {
            "type": "named",
            "name": "bigint"
          }
        }
      },
      "update_operators": {}
    },
    "character varying": {
      "aggregate_functions": {},
      "comparison_operators": {
        "_eq": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_gt": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_gte": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_ilike": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_iregex": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_like": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_lt": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_lte": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_neq": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_nilike": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_niregex": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_nlike": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_nregex": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_nsimilar": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_regex": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "_similar": {
          "argument_type": {
            "type": "named",
            "name": "character varying"
          }
        }
      },
      "update_operators": {}
    },
    "numeric": {
      "aggregate_functions": {},
      "comparison_operators": {
        "_eq": {
          "argument_type": {
            "type": "named",
            "name": "numeric"
          }
        },
        "_gt": {
          "argument_type": {
            "type": "named",
            "name": "numeric"
          }
        },
        "_gte": {
          "argument_type": {
            "type": "named",
            "name": "numeric"
          }
        },
        "_lt": {
          "argument_type": {
            "type": "named",
            "name": "numeric"
          }
        },
        "_lte": {
          "argument_type": {
            "type": "named",
            "name": "numeric"
          }
        },
        "_neq": {
          "argument_type": {
            "type": "named",
            "name": "numeric"
          }
        }
      },
      "update_operators": {}
    },
    "timestamp without time zone": {
      "aggregate_functions": {},
      "comparison_operators": {
        "_eq": {
          "argument_type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "_gt": {
          "argument_type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "_gte": {
          "argument_type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "_lt": {
          "argument_type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "_lte": {
          "argument_type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "_neq": {
          "argument_type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        }
      },
      "update_operators": {}
    }
  },
  "object_types": {
    "Album": {
      "fields": {
        "AlbumId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "ArtistId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Title": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        }
      }
    },
    "Artist": {
      "fields": {
        "ArtistId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Name": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        }
      }
    },
    "Customer": {
      "fields": {
        "Address": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "City": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Company": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Country": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "CustomerId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Email": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Fax": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "FirstName": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "LastName": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Phone": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "PostalCode": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "State": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "SupportRepId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        }
      }
    },
    "Employee": {
      "fields": {
        "Address": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "BirthDate": {
          "type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "City": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Country": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Email": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "EmployeeId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Fax": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "FirstName": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "HireDate": {
          "type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "LastName": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Phone": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "PostalCode": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "ReportsTo": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "State": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "Title": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        }
      }
    },
    "Genre": {
      "fields": {
        "GenreId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Name": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        }
      }
    },
    "Invoice": {
      "fields": {
        "BillingAddress": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "BillingCity": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "BillingCountry": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "BillingPostalCode": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "BillingState": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "CustomerId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "InvoiceDate": {
          "type": {
            "type": "named",
            "name": "timestamp without time zone"
          }
        },
        "InvoiceId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Total": {
          "type": {
            "type": "named",
            "name": "numeric"
          }
        }
      }
    },
    "InvoiceLine": {
      "fields": {
        "InvoiceId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "InvoiceLineId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Quantity": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "TrackId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "UnitPrice": {
          "type": {
            "type": "named",
            "name": "numeric"
          }
        }
      }
    },
    "MediaType": {
      "fields": {
        "MediaTypeId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Name": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        }
      }
    },
    "Playlist": {
      "fields": {
        "Name": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "PlaylistId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        }
      }
    },
    "PlaylistTrack": {
      "fields": {
        "PlaylistId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "TrackId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        }
      }
    },
    "Track": {
      "fields": {
        "AlbumId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Bytes": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Composer": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "GenreId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "MediaTypeId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Milliseconds": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "Name": {
          "type": {
            "type": "named",
            "name": "character varying"
          }
        },
        "TrackId": {
          "type": {
            "type": "named",
            "name": "bigint"
          }
        },
        "UnitPrice": {
          "type": {
            "type": "named",
            "name": "numeric"
          }
        }
      }
    }
  },
  "collections": [
    {
      "name": "Album",
      "arguments": {},
      "type": "Album",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Album": {
          "unique_columns": [
            "AlbumId"
          ]
        }
      },
      "foreign_keys": {
        "FK_AlbumArtistId": {
          "column_mapping": {
            "ArtistId": "ArtistId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    },
    {
      "name": "Artist",
      "arguments": {},
      "type": "Artist",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Artist": {
          "unique_columns": [
            "ArtistId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "Customer",
      "arguments": {},
      "type": "Customer",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Customer": {
          "unique_columns": [
            "CustomerId"
          ]
        }
      },
      "foreign_keys": {
        "FK_CustomerSupportRepId": {
          "column_mapping": {
            "SupportRepId": "EmployeeId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    },
    {
      "name": "Employee",
      "arguments": {},
      "type": "Employee",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Employee": {
          "unique_columns": [
            "EmployeeId"
          ]
        }
      },
      "foreign_keys": {
        "FK_EmployeeReportsTo": {
          "column_mapping": {
            "ReportsTo": "EmployeeId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    },
    {
      "name": "Genre",
      "arguments": {},
      "type": "Genre",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Genre": {
          "unique_columns": [
            "GenreId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "Invoice",
      "arguments": {},
      "type": "Invoice",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Invoice": {
          "unique_columns": [
            "InvoiceId"
          ]
        }
      },
      "foreign_keys": {
        "FK_InvoiceCustomerId": {
          "column_mapping": {
            "CustomerId": "CustomerId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    },
    {
      "name": "InvoiceLine",
      "arguments": {},
      "type": "InvoiceLine",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_InvoiceLine": {
          "unique_columns": [
            "InvoiceLineId"
          ]
        }
      },
      "foreign_keys": {
        "FK_InvoiceLineInvoiceId": {
          "column_mapping": {
            "InvoiceId": "InvoiceId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        },
        "FK_InvoiceLineTrackId": {
          "column_mapping": {
            "TrackId": "TrackId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    },
    {
      "name": "MediaType",
      "arguments": {},
      "type": "MediaType",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_MediaType": {
          "unique_columns": [
            "MediaTypeId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "Playlist",
      "arguments": {},
      "type": "Playlist",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Playlist": {
          "unique_columns": [
            "PlaylistId"
          ]
        }
      },
      "foreign_keys": {}
    },
    {
      "name": "PlaylistTrack",
      "arguments": {},
      "type": "PlaylistTrack",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_PlaylistTrack": {
          "unique_columns": [
            "PlaylistId",
            "TrackId"
          ]
        }
      },
      "foreign_keys": {
        "FK_PlaylistTrackPlaylistId": {
          "column_mapping": {
            "PlaylistId": "PlaylistId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        },
        "FK_PlaylistTrackTrackId": {
          "column_mapping": {
            "TrackId": "TrackId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    },
    {
      "name": "Track",
      "arguments": {},
      "type": "Track",
      "deletable": false,
      "uniqueness_constraints": {
        "PK_Track": {
          "unique_columns": [
            "TrackId"
          ]
        }
      },
      "foreign_keys": {
        "FK_TrackAlbumId": {
          "column_mapping": {
            "AlbumId": "AlbumId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        },
        "FK_TrackGenreId": {
          "column_mapping": {
            "GenreId": "GenreId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        },
        "FK_TrackMediaTypeId": {
          "column_mapping": {
            "MediaTypeId": "MediaTypeId"
          },
          "foreign_collection": "information_schema.constraint_table_usage is empty in cockroach"
        }
      }
    }
  ],
  "functions": [],
  "procedures": []
}
